### DISCUSSION TITLE
Is ionization current implemented inside WarpX?

### DISCUSSION THREAD

**Discussion by GivenChen** [BLAST-WarpX/warpx#4963] [Q&A] [answered]
Hi,
I wonder is ionization current implemented inside WarpX? I tried to look for ionization current inside documentation and couldn't find it.  When an ionization event happens, an ionization current should be considered to reduce the electric field to conserve the energy, because some EM energy is spent on ionizing the electrons.

**Comment by RemiLehe** [comment, marked as answer]
Thanks for the question.
Yes, you are right: in order to take into account the field energy that is consumed when ionizing an atom, an ionization current should be added.
This is currently **not** included in WarpX. In other words, in WarpX the ionization current is neglected.
We could consider adding it, if you have a use case where this effect is non-negligible.

**Reply by GivenChen** [reply]
Hi Remi,

Yes, I am considering a case where ionized carriers alter the pulse frequency, which is essentially-- some photons in one photon out, and the energy conservation would be important in this scenario. I would really appreciated it if the ionization current could be implemented in the near future.

Thank you~
